eZ/Publish/Core/Persistence/Cache/UrlWildcardHandler.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+namespace eZ\Publish\Core\Persistence\Cache;
+
+use eZ\Publish\API\Repository\Exceptions\NotFoundException as APINotFoundException;
+use eZ\Publish\Core\Base\Exceptions\NotFoundException;
+use eZ\Publish\SPI\Persistence\Content\UrlWildcard;
+use eZ\Publish\SPI\Persistence\Content\UrlWildcard\Handler as UrlWildcardHandlerInterface;
+
+class UrlWildcardHandler extends AbstractHandler implements UrlWildcardHandlerInterface
+{
+    /**
+     * Constant used for storing not found results for lookup().
+     */
+    const NOT_FOUND = 0;
+
+    /**
+     * @see \eZ\Publish\SPI\Persistence\Content\UrlWildcard\Handler::create
+     */
+    public function create($sourceUrl, $destinationUrl, $forward = false)
+    {
+        $this->logger->logCall(
+            __METHOD__,
+            [
+                'sourceUrl' => $sourceUrl,
+                'destinationUrl' => $destinationUrl,
+                'forward' => $forward,
+            ]
+        );
+
+        $urlWildcard = $this->persistenceHandler->urlWildcardHandler()->create($sourceUrl, $destinationUrl, $forward);
+
+        $this->cache->invalidateTags(['urlWildcard-notFound']);
+
+        return $urlWildcard;
+    }
+
+    /**
+     * @see \eZ\Publish\SPI\Persistence\Content\UrlWildcard\Handler::remove
+     */
+    public function remove($id)
+    {
+        $this->logger->logCall(__METHOD__, ['id' => $id]);
+
+        $urlWildcard = $this->load($id);
+
+        $this->persistenceHandler->urlWildcardHandler()->remove($urlWildcard->id);
+
+        // need to clear lists of UrlWildcards cached due to loadAll()
+        $this->cache->invalidateTags(['ez-urlWildcard-id-' . $urlWildcard->id]);
+    }
+
+    /**
+     * @see \eZ\Publish\SPI\Persistence\Content\UrlWildcard\Handler::load
+     */
+    public function load($id)
+    {
+        $this->logger->logCall(__METHOD__, ['id' => $id]);
+
+        $cacheItem = $this->cache->getItem('ez-urlWildcard-id-' . $id);
+
+        if ($cacheItem->isHit()) {
+            if (($return = $cacheItem->get()) === self::NOT_FOUND) {
```

Comment:
wrong copy paste from translate right? as in on id lookup we don't need to cache missing items I would assume?

Reply:
Indeed, good catch, thanks!